pkg/applicationserver/io/mqtt/mqtt.go

```diff
@@ -252,6 +266,13 @@ func (c *connection) Connect(ctx context.Context, info *auth.Info) (context.Cont
 		return nil, err
 	}
 	ctx = c.io.Context()
+
+	authTokenID := ""
+	if _, v, _, err := ttsauth.SplitToken(string(info.Password)); err != nil && v != "" {
+		authTokenID = v
+	}
+	c.resource = ratelimit.ApplicationMQTTDownResource(ctx, ids, authTokenID)
```

Comment:
So if `authTokenID` is empty, the rate limiting pool is shared?

What if the client is DoSing with a random password?

Shouldn't we hash the password instead, if splitting the token doesn't work?

Reply:
This is only ever reached if `c.server.Subscribe(ctx, "mqtt", &ids, true)` succeeds, which means that the client has supplied valid auth.